### PR TITLE
Filter debugging

### PR DESCRIPTION
**Pull request by vitvly**
Created to verify https://github.com/status-im/status-go/pull/4020

Tracking issue: https://github.com/status-im/status-go/issues/4042

UPDATE: ~~this has some build issues related to github.com/waku-org/go-zerokit-rln-apple i think, working on it.~~ Fixed

### CODE REVIEW

File: status-go-version.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "_comment": "Instead use: scripts/update-status-go.sh <rev>",
     "owner": "status-im",
     "repo": "status-go",
-    "version": "v0.166.11",
-    "commit-sha1": "6bcf5f1289f9160168574290cbd6f90dede3f8f6",
-    "src-sha256": "0g56pawjpxgjp93slvpbkajz73ajrjfi0n0js3hvqh8x4ciwa43p"
+    "version": "4bd6eb6efe00c55ceb8",
```
Comment (siddarthkay) [inline status-go-version.json]: not sure but version seems a bit wonky, you can point to your branch instead like this : 
`scripts/update-status-go.sh filter-debugging`

**Comment by vitvly** [comment]
Verifying e2e tests.

**Review by jakubgs — APPROVED** [review]
Just remember to tag the `status-go` version.

**Comment by churik** [comment]
@siphiuel 
currently messaging is not working between 2 mobile devices - it can be your commit or any commit between 0.167.2 and your branch

1) "Contact request accepted" is not received (Android).
Received after forcing app to restart.
2) User sending messages - no message in  UI, and at logout I could see this error
IMAGE 2023-09-25 14:01:40  (Android)

Update: after re-login received "Contact request accepted" and can send the messages after that
3) Endless logout on IOS
Logs (Android):
[Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/12714801/Status-debug-logs.zip)
Logs (IOS):
[logs (60).zip](https://github.com/status-im/status-mobile/files/12714806/logs.60.zip)

**Comment by vitvly** [comment]
@churik thank you! Also testing this manually. Indeed there have been lots of commits since 0.167.2. I'll check the version just prior to filter commits first, to be sure whether these issues are related to filter refactor, and then go on from there.

**Comment by vitvly** [comment]
Fleet nodes return `EOF` on `IsSubscriptionAlive` calls, and libp2p's `SupportsProtocols` calls do not return filter protocol in the list (https://github.com/status-im/status-mobile/issues/16873#issuecomment-1734439024)

**Comment by vitvly** [comment]
Seems that liveness checks also fail when running status-go tests: https://github.com/status-im/status-mobile/issues/16873#issuecomment-1735772369

**Comment by vitvly** [comment]
Updated with latest go-waku, more detail here: https://github.com/status-im/status-mobile/issues/16873#issuecomment-1735949822

**Comment by vitvly** [comment]
Able to receive "Contact request accepted" messages and to chat afterwards, but logout still hanging. Fixing. Also checking e2e test failures.

**Review by siddarthkay — APPROVED** [review]

**Comment by vitvly** [comment]
Superseded by https://github.com/status-im/status-mobile/pull/17472

**Pull request closed without merging**